server/channels/api4/scheduled_post.go

```diff
@@ -52,3 +53,38 @@ func createSchedulePost(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 }
+
+func getTeamScheduledPosts(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireTeamId()
+	if c.Err != nil {
+		return
+	}
+
+	teamId := c.Params.TeamId
+	userId := c.AppContext.Session().UserId
+
+	scheduledPosts, appErr := c.App.GetUserTeamScheduledPosts(c.AppContext, userId, teamId)
+	if appErr != nil {
+		c.Err = appErr
+		return
+	}
+
+	response := map[string][]*model.ScheduledPost{}
+	response[teamId] = scheduledPosts
+
+	if r.URL.Query().Get("includeDirectChannels") == "true" {
+		directChannelScheduledPosts, appErr := c.App.GetUserTeamScheduledPosts(c.AppContext, userId, "")
+		if appErr != nil {
+			c.Err = appErr
+			return
+		}
+
+		response["directChannels"] = directChannelScheduledPosts
```

Comment:
I though about this and empty string doesn't feel right. Prefer to keep this.

Reply:
It might feel odd (especially since you'd have to access the state as `scheduledPosts['']`), but even if it feels weird, I feel like it's a case where consistency would be good. I'm only maybe 2/5 on that though.

That said, and this sounds unintentionally petty to say after potentially unsuccessfully convincing you to change to the empty string, but this should probably be named `direct_channels` to fit our naming convention of using snake case for JSON. We really should've ditched that naming convention years ago